cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -137,7 +138,8 @@
   private final SchedulerQueueResolver schedulerQueueResolver;
 
   // Handles that are running, or not yet completely fetched, they have longer timeout
-  private final Cache<QueryHandle, OperationInfo> activeHandleCache;
+  @VisibleForTesting
+  public final Cache<QueryHandle, OperationInfo> activeHandleCache;
```

Comment:
Until the query is closed, it remains in the active handler cache. So you have control on when it gets removed from the cache. The test will not be flaky.